module-4/backend/week_four.md

```diff
@@ -1,11 +1,58 @@
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
-TBD  
 
-#### Review  
+# Week Four - Module 4 Recap
 
-TBD
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
+
+Note: When you're done, submit a PR.
+
+    What's the purpose of a project management tool?
+      To keep track of a project: what is done, what is being worked on, what needs to be worked on, who is working on what.
+    What are the stages of a card when using Pivotal Tracker?
+      gathering requirements
+      story breakdown
+      story estimation
+      prioritization
+      coding out the story
+      story acceptance
+
+    What are the four types of cards available in Pivotal Tracker?
+      current stories
+      icebox
+      backlog
+      done
+    What's your favorite project management tool?
+      Waffle
+    Why do we create staging environments?
+      So that we can make sure everything is working properly in fake production before putting it to actual production
+    What are the characteristics of a good README (in your opinion)?
+      Have a setup, a quick start, good examples of how to program is used, what the program does, possible endpoints, notes on things that could be difficult, and all docs are explained as if a beginner is reading them.
+    What's one main improvement you're going to make to your code regarding accessibility issues?
+      make this tab-able
+    What are some basic security concerns to be aware of when building applications?
+      making sure you can't access content you should when you log in
+      password storage
+      unencrypted traffic
+      search bar and url - sql injectable
+      faulty authentication
+    Why is continuous integration helpful/important?
+      so that as you build you continue to make the iteration loop - make, refactor, review, redo - and so as you go your making the code better now waiting til the end.
+    What are some continuous integration tools?
+      breaking the project or problem down to base functionality and building onto it from there with the next possible piece
+
+# Review
+    What are some characteristics of "good" git workflow?
+        commiting often, merging branches back to master not just leaving them out to dry, push when you stop coding for the night
+    What are the four fundamental concepts of object oriented programming?
+        encapsulation
+        abstraction
+        inheritance
+        polymorphism
+
+    What's a module in Ruby and what's the difference between a class and a module?
+      A module is like a class because it has methods and performs actions but classes can inherit from a module, and when that happens its like all the code from that file is pasted in the file that is inheriting the code
```

Comment:
classes do not inherit from a module. a module is included in a class. this is a pretty significant difference. a module encapsulates a set of behaviors or methods that can be mixed into a class. a class encapsulates behavior and state